### ISSUE TITLE
Increasing batch_size increases training time

### ISSUE THREAD

**Issue by xstha** [ostris/ai-toolkit#286]
## This is for bugs only

Did you already ask [in the discord](https://discord.gg/VXmU2f5WEU)?

Yes

You verified that this is a bug and not a feature request or question by asking [in the discord](https://discord.gg/VXmU2f5WEU)?

Yes

## Describe the bug

Training is way slower than expected on a 94GB VRAM GPU.

Using 40 images (1152x2048), 4000 steps, batch size 4.

Increasing batch size from 2 to 4 doubles the training time instead of speeding it up.

With batch_size: 2:
my_flux_lora: 0%| | 3/4000 [00:34<9:22:56, 8.45s/it, lr: 4.0e-04 loss: 4.564e-01]

With batch_size: 4:
my_flux_lora: 0%| | 1/4000 [00:33<18:25:01, 16.58s/it, lr: 4.0e-04 loss: 4.384e-01]

Only ~23% VRAM is in use. Also, linear=2, linear_alpha=16 are set, but logs say: create LoRA network. base dim (rank): 2, alpha: 2 — shouldn't it be 16?

Is this expected behaviour?

I later tried setting `batch_size ` to 1 which reduced training time to 2.5hrs, but only 23% of the VRAM was in use, so it's not fully utilizing the resources.

config.yaml:

```
job: extension
config:
  name: my_flux_lora
  process:
    - type: sd_trainer
      training_folder: /root/ai-toolkit/modal_output
      performance_log_every: 1000
      device: cuda:0
      trigger_word: my_flux_lora
      network:
        type: lora
        linear: 2
        linear_alpha: 16
      save:
        dtype: float16
        save_every: 400
        max_step_saves_to_keep: 4
      datasets:
        - folder_path: /root/ai-toolkit/datasets/my_flux_lora
          caption_ext: txt
          caption_dropout_rate: 0.05
          shuffle_tokens: false
          cache_latents_to_disk: true
          resolution: [1152]
      train:
        batch_size: 4
        steps: 4000
        gradient_accumulation_steps: 1
        train_unet: true
        train_text_encoder: false
        gradient_checkpointing: true
        noise_scheduler: flowmatch
        optimizer: adamw8bit
        lr: 0.0004
        lr_scheduler: cosine_with_restarts
        ema_config:
          use_ema: true
          ema_decay: 0.99
        dtype: bf16
      model:
        name_or_path: black-forest-labs/FLUX.1-dev
        is_flux: true
        quantize: true
      sample:
        sampler: flowmatch
        sample_every: 400
        width: 720
        height: 1280
        prompts:
          - '[trigger],...'
        neg: ''
        seed: 42
        walk_seed: true
        guidance_scale: 3
        sample_steps: 20
meta:
  name: my_flux_lora
  version: '1.0'
```

**Comment by Cinkye** [comment]
Seems that you need to adjust the step. If you change batch_size from 2 to 4, steps should be 0.5x, in that case, 2000.
